pkg/kds/v2/server/event_based_watchdog.go

```diff
@@ -0,0 +1,83 @@
+package server
+
+import (
+	"context"
+	"time"
+
+	envoy_core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	"github.com/go-logr/logr"
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/kumahq/kuma/pkg/core"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+	"github.com/kumahq/kuma/pkg/events"
+	"github.com/kumahq/kuma/pkg/kds/v2/reconcile"
+	"github.com/kumahq/kuma/pkg/multitenant"
+	util_watchdog "github.com/kumahq/kuma/pkg/util/watchdog"
+)
+
+type EventBasedWatchdog struct {
+	Ctx                  context.Context
+	Node                 *envoy_core.Node
+	StreamID             int64
```

Comment:
Why do we need `StreamID` here? I don't see it being used anywhere

Reply:
Removed